src/corva/utils.py

```diff
@@ -0,0 +1,69 @@
+from typing import Any, Optional
```

Comment:
Can we put these functions into an appropriate class or module? A utils.py file usually ends up just being spaghetti after a while.